### PR TITLE
30-m1-create-function-to-create-errors: function to create errors and…

### PR DESCRIPTION
**Pull request by CamiloBorba09**
… error types have been added

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
-    "tb" : "Integer",
-    "v" : "Integer"
-}
+export const errorTypes = {
+  "unknown-key": "Key is not part of reserved keys",
+  "invalid-value": "Value does not meet requirements",
+  "wrong-type-value": "Value type is incorrect",
+  "incorrect-format": "Format is incorrect",
+  "parameter-encoding": "Parameter is not encoded",
+  "missing-key": "Key is missed",
```
Comment (arrivillagaTomas) [inline src/constants.js:28]: "Key is missing"

File: src/error.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { errorTypes } from "./constants.js";
+
+export const createError = (type, key, value) => {
+  if (typeof type != "string" || errorTypes.hasOwnProperty(type) == false) {
```
Comment (arrivillagaTomas) [inline src/error.js:4]: if (typeof type !== 'string' || !errorTypes[type])

**Review by arrivillagaTomas — COMMENTED** [review]
Looks good,  not sure what changed for the keyTypes

File: src/error.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { errorTypes } from "./constants.js";
+
+export const createError = (type, key, value) => {
+  if (typeof type !== "String" || !errorTypes[type]) {
+    console.error("Type not defined");
```
Comment (mcamardon) [inline src/error.js:5]: Maybe to make it clearer, change the message to "Error type not defined."

File: src/error.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { errorTypes } from "./constants.js";
+
+export const createError = (type, key, value) => {
+  if (typeof type !== "String" || !errorTypes[type]) {
```
Comment (mcamardon) [inline src/error.js:4]: Is it necessary to check if the type is String? I think it could work without it.

Comment (CamiloBorba09) [reply src/error.js]: I checked it and it is not necessary. Thanks!

**Review by mcamardon — APPROVED** [review]

**Review by arrivillagaTomas — APPROVED** [review]

**Review by sebastianpiq — APPROVED** [review]

**Pull request merged**